velox/functions/prestosql/tests/ReduceTest.cpp

```diff
@@ -125,7 +125,7 @@ TEST_F(ReduceTest, finalSelection) {
       "reduce(c0, 10, (s, x) -> s + x, s -> row_constructor(s)))",
       input);
 
-  auto expectedResult = makeRowVector({makeFlatVector<int64_t>(
+  auto expectedResult = makeFlatVector<int64_t>(
```

Comment:
nit: can you revert these unrelated changes

Reply:
Sorry seems I pushed a dirty commit by mistake.

Reply:
Not actually. After changes in `Reduce` function, this `RowType` output is different from the result. The inner layer is same but for `RowType`, the expected one is like `{null}, {1}, {2}....` while actual result is `null, {1}, {2}....`

Reply:
@yma11 `null, {1}, {2}.... ` is what should be expected. evaluate<> returns the result of the expression which should have the null at the top level rowVector layer and this change to check the child of the row is actually hiding this bug. The expression used in the test  `if (c1 < 100, row_constructor(c1), reduce(c0, 10, (s, x) -> s + x, s -> row_constructor(s)))` adds a top level null to the result via the first branch  `row_constructor(c1)` which means the mismtach` {null} != null ` is a valid bug. This is happening because in this change the following piece of code is overwriting the row in Reduce which should not have been touched. 
```
   for (int i = 0; i < flatArray->size(); i++) {
      if (flatArray->isNullAt(i)) {
        result->setNull(i, true);
      }
    }
```
Please see explanation of why this is happening in my other comment.
Therefore, to fix this test we need to add top level nulls to  expectedResult at the RowVector layer.

Reply:
Done.